virttest/virt_vm.py

```diff
@@ -684,93 +683,48 @@ def _get_address(self, index=0, ip_version="ipv4"):
-                    raise VMIPAddressMissingError(mac, ip_version)
-                macs = self.virtnet.mac_list()
-                reachable = utils_net.verify_ip_address_ownership(
-                    ip_addr, macs, timeout=16, devs=devs)
-                if not reachable:
-                    raise VMAddressVerificationError(mac, ip_addr)
-            except VMAddressVerificationError:
-                nic_params = self.params.object_params(nic.nic_name)
-                pci_assignable = nic_params.get("pci_assignable") != "no"
-                # SR-IOV/Macvtap cards may not be in same subnet with the cards
-                # used by host by default, so arp checks won't work. Therefore,
-                # do not raise VMAddressVerificationError when SR-IOV is used.
-                if pci_assignable or nic.nettype == "macvtap":
-                    nic_backend = pci_assignable and "SR-IOV" or "macvtap"
-                    msg = "Could not verify DHCP lease: %s-> %s." % (nic.mac,
-                                                                     ip_addr)
-                    msg += " Maybe %s is not in the same subnet " % ip_addr
-                    msg += "as the host (%s in use)" % nic_backend
-                    logging.error(msg)
-                else:
-                    if mac in self.address_cache:
-                        self.address_cache[mac] = None
-                    else:
-                        # clean item in host arp table
-                        os.system("arp -d %s" % ip_addr)
-                    logging.debug("Clean up outdated address info, "
-                                  "MAC: %s, IP: %s" % (mac, ip_addr))
-                    raise
-            if not ip_addr:
-                raise VMIPAddressMissingError(mac, ip_version)
-            # Update address_cache since ip/mac releation ship verify pass
-            self.address_cache[mac] = ip_addr
-            return ip_addr
-        else:
-            # Try to get and return IPV6 address
-            if self.params.get('using_linklocal') == "yes":
-                ipv6_addr = utils_net.ipv6_from_mac_addr(nic.mac)
-            # Using global address
-            else:
-                mac_key = "%s_6" % nic.mac
-                ipv6_addr = self.address_cache.get(mac_key.lower())
-            if not ipv6_addr:
-                raise VMIPAddressMissingError(nic.mac, ip_version)
-            # Check whether the ipv6 address is reachable
-            utils_net.refresh_neigh_table(nic.netdst, ipv6_addr)
-            if not utils_misc.wait_for(lambda: utils_net.neigh_reachable(
-                                       ipv6_addr, nic.netdst),
-                                       30, 0, 1, "Wait neighbour reachable"):
-                raise VMAddressVerificationError(nic.mac, ipv6_addr)
-            return ipv6_addr
+        mac_pattern = "%s_6" if ip_version == "ipv6" else "%s"
+        ip_addr = self.address_cache.get(mac_pattern % mac)
```

Comment:
I am not sure `address_cache` got declared ?

Reply:
It is declared in child classes, see `qemu_vm`, `libvirt_vm` and `ovirt`.

I was planned to fix this, but it was a little bit nasty as well, so just leave it for next fix.

Reply:
okay, lets have a `# Todo:` for our reference